### DISCUSSION TITLE
Black and white when using docker

### DISCUSSION THREAD

**Discussion by tuxtlequino** [charmbracelet/wish#383] [Q&A]
I have this program that works well, but I needed to put it in a container. The problem is that all I get now is black and white. I have tried a lot of different things (including changing the os.Setenv to different values). I will include my main.go and the dockerfile if that helps.
```go
package main

import (
	"fmt"
	"order/tui"
	"os"

	tea "github.com/charmbracelet/bubbletea"
	"github.com/charmbracelet/log"
	ssh "github.com/charmbracelet/ssh"
	wish "github.com/charmbracelet/wish"
	wishbubbletea "github.com/charmbracelet/wish/bubbletea"
	wishlog "github.com/charmbracelet/wish/logging"
	"github.com/muesli/termenv"
)

const validPassword = "NOTVALID;"

var users = map[string]string{

	// You can add add your name and public key here :)
}

func main() {
	os.Setenv("TERM", "screen-256color")

	server, err := wish.NewServer(
		wish.WithAddress(":2223"),
		wish.WithHostKeyPath(".ssh/term_info_ed25519"),
		// First, public-key authentication:
		wish.WithPublicKeyAuth(func(_ ssh.Context, key ssh.PublicKey) bool {
			log.Info("public-key")
			for _, pubkey := range users {
				parsed, _, _, _, _ := ssh.ParseAuthorizedKey(
					[]byte(pubkey),
				)
				if ssh.KeysEqual(key, parsed) {
					return true
				}
			}
			return false
		}),

		// Then, password.
		wish.WithPasswordAuth(func(_ ssh.Context, password string) bool {
			log.Info("password")
			return password == validPassword
		}), wish.WithMiddleware(
			wishlog.Middleware(),
			wishbubbletea.MiddlewareWithColorProfile(sessionHandler, termenv.ANSI),
			// wishbubbletea.Middleware(sessionHandler),
		),
	)
	if err != nil {
		fmt.Println("Failed to create SSH server:", err)
		os.Exit(1)
	}

	fmt.Println("Starting SSH server on port 2223...")
	err = server.ListenAndServe()
	if err != nil {
		fmt.Println("Failed to start SSH server:", err)
		os.Exit(1)
	}
}

func sessionHandler(s ssh.Session) (tea.Model, []tea.ProgramOption) {

	// Create the main model
	mainModel := tui.NewMainModelSSH(s)

	// Return the model and program options
	return mainModel, []tea.ProgramOption{
		tea.WithInput(s),
		tea.WithOutput(s),
		tea.WithAltScreen(), // Use the alternate screen buffer
		// tea.WithMouseCellMotion(), // Enable mouse support
	}
}
```
Dockerfile
```
# -------------------------------------------------------
# 1. Build Stage (Go only)
# -------------------------------------------------------
FROM golang:1.23.4 AS builder

# We only need minimal packages to build Go (e.g., git).
# Do NOT install Python here unless you truly need it during build.
RUN apt-get update && \
    apt-get install -y git && \
    rm -rf /var/lib/apt/lists/*

# Set the working directory
WORKDIR /app

# Copy Go modules and download dependencies first
COPY go.mod go.sum ./
RUN go mod download

# Copy the rest of your application source code
COPY . .

# Build your Go binary
RUN go build -o main .

# -------------------------------------------------------
# 2. Final Runtime Stage
# -------------------------------------------------------
FROM debian:stable-slim

# Install Python + venv
RUN apt-get update && \
    apt-get install -y python3 python3-pip python3-venv && \
    rm -rf /var/lib/apt/lists/*

# Create a Python virtual environment in /venv
RUN python3 -m venv /venv

# Ensure any pip/python calls use the venv
ENV PATH="/venv/bin:$PATH"

# Install PyPDF2 (and other Python libraries you need)
RUN pip install --no-cache-dir PyPDF2

# Create an /app folder for your runtime
WORKDIR /app

# Copy the compiled Go binary from the builder stage
COPY --from=builder /app/main /app/main

# If you have a config folder in your source, copy it in
# (adjust path as needed)
COPY config /app/config

# Environment variables
.....
....

# Expose an SSH port or whatever your app needs
EXPOSE 2223

# Finally, run your Go binary
CMD ["./main"]
```

**Comment by lucsoft** [comment]
Anyone fixed this issue?

**Comment by caarlos0** [comment]
there's some explanation in https://github.com/charmbracelet/wish/pull/273/files

unfortunately its not merged, will work on it.

**Reply by caarlos0** [reply]
also worth noting that this should be fixed in v2 hopefully

**Reply by caarlos0** [reply]
PS: trick is to use a custom lipgloss renderer

see https://pkg.go.dev/github.com/charmbracelet/wish@v1.4.4/bubbletea#MakeRenderer

**Comment by tuxtlequino** [comment]
Okay. Do you mean, this will work on wish version 2? I also tried to understand the trick to use a custom lipgloss renderer, but I couldn't figure it out. Is there an example of this somewhere? For example, the wish program I have starts several models from a main model. This is the first one but I have no clue how would I apply a custom renderer there. Thanks!

package tui

import (
	"github.com/charmbracelet/bubbles/list"
	tea "github.com/charmbracelet/bubbletea"
)

// Define your service types
var functionType = []struct {
	Name string
	Kind string
}{
	{"First Option", "Order"},
	{"Second Option", "Booklet"},
}

type selectFunctionModel struct {
	list   list.Model
	width  int
	height int
	err    error
}

func NewSelectFunctionModel() *selectFunctionModel {
	items := make([]list.Item, len(functionType))
	for i, s := range functionType {
		items[i] = listItem(s.Name) // Use shared listItem type
	}
	ls := list.New(items, list.NewDefaultDelegate(), 0, 0)
	ls.Title = "What would you like to do?"
	ls.SetFilteringEnabled(false)
	ls.SetShowStatusBar(false)
	ls.SetHeight(len(functionType)*3 + 5)
	return &selectFunctionModel{
		list: ls,
	}
}

func (m *selectFunctionModel) Init() tea.Cmd {
	return nil
}


func (m *selectFunctionModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
	switch msg := msg.(type) {

	case tea.WindowSizeMsg:
		m.width = msg.Width
		m.height = msg.Height
		m.list.SetWidth(msg.Width)

		return m, nil

	case SetSizeMsg:
		m.width = msg.Width
		m.height = msg.Height
		m.list.SetWidth(msg.Width)
		return m, nil

	case tea.KeyMsg:
		switch msg.String() {
		case "enter":
			i := m.list.Index()
			if i < 0 || i >= len(functionType) {
				return m, nil
			}
			svc := functionType[i]
			return m, func() tea.Msg {
				return SelectedMainFuncMsg{
					Functionality: svc.Kind,
				}
			}
		case "q", "ctrl+c":
			return m, tea.Quit
		}
	}

	var cmd tea.Cmd
	m.list, cmd = m.list.Update(msg)
	return m, cmd
}

func (m *selectFunctionModel) View() string {
	return CenterHorizontally(m.list.View(), m.width)
}

**Reply by caarlos0** [reply]
there are a couple of them https://github.com/search?q=repo%3Acharmbracelet%2Fwish%20makerenderer&type=code

you'll need to make the renderer, pass it down to your models, and use it to create your styles.

bubbletea and lipgloss v2 will work fundamentally differently in that regard, and this shouldn't be necessary anymore then. We have not set a timeframe for its release yet, though we are working on it.

**Comment by tuxtlequino** [comment]
Thank you. That is what I feared. At this point it seems it would require a lot of work all over my code since I have to redefine all the styles in the bubble models, unless I am missing something obvious. I also tried just with the v2 and I cannot get pass another error. I will try to create another issue for that. Again, thank you!

**Reply by caarlos0** [reply]
please note that v2 is not ready yet, and there is no compatible wish version with bubbletea v2, which is under heavy development.

**Comment by caarlos0** [comment]
FWIW, I pushed github.com/charmbracelet/wish/v2@v2exp - this uses all the new libraries, if anyone wants to try it out.

example bubbletea app: https://github.com/charmbracelet/wish/blob/v2-exp/examples/bubbletea/main.go

Notice there's no need of `MakeRenderer` anymore.

See https://github.com/charmbracelet/wish/pull/392
